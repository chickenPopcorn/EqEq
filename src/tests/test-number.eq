SomeCtx = { a = { 42; } }

SomeCtx:find a {
  print("%0.0f\n", a);
  print("%0.5f\n", 4.2e-4 );
  print("%0.2f\n", 0.42   );
  print("%0.2f\n", .42    );
  print("%0.1f\n", 0e1    );
  print("%0.0f\n", 4.2e4  );
  print("%0.0f\n", -42    );
  print("%0.5f\n", -4.2e-4);
  print("%0.2f\n", -0.42  );
  print("%0.2f\n", -.42   );
  print("%0.1f\n", -0e1   );
  print("%0.0f\n", -4.2e4 );
  print("%f\n", 5 * -4.2e-4);
}
