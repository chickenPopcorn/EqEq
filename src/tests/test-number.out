42
0.00042
0.42
0.42
0.0
42000
-42
-0.00042
-0.42
-0.42
-0.0
-42000
-0.002100
